master/pkg/searcher/custom_search.go

```diff
@@ -0,0 +1,133 @@
+package searcher
+
+import (
+	"encoding/json"
+
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/master/pkg/schemas/expconf"
+	"github.com/determined-ai/determined/proto/pkg/experimentv1"
+)
+
+type (
+	customSearchState struct {
+		// store the operations
+		// store the events
+		SearchMethodType   SearchMethodType `json:"search_method_type"`
+		SearcherEventQueue *SearcherEventQueue
+	}
+
+	customSearch struct {
+		defaultSearchMethod
+		expconf.CustomConfig
+		customSearchState
+	}
+)
+
+func newCustomSearch(config expconf.CustomConfig) SearchMethod {
+	return &customSearch{
+		CustomConfig: config,
+		customSearchState: customSearchState{
+			SearchMethodType:   CustomSearch,
+			SearcherEventQueue: createSearcherEventQueue(),
+		},
+	}
+}
+
+func createSearcherEventQueue() *SearcherEventQueue {
+	return newSearcherEventQueue()
+}
+func (s *customSearch) initialOperations(ctx context) ([]Operation, error) {
+	event := experimentv1.SearcherEvent_InitialOperations{
+		InitialOperations: &experimentv1.InitialOperations{},
+	}
+	searcherEvent := experimentv1.SearcherEvent{
+		Id:    -1,
```

Comment:
Oh right, I thought of that issue but then forgot about it, sorry. Yeah, so it's awkward because the interface type of the `Event` field is not exported in the generated protobuf code, for some unknown reason. (I thought it was to ensure that no external code could implement the interface for an unexpected type, but I believe that's still guaranteed if the interface type is exported and the method it contains is not.) I was imagining `Enqueue` would just take `any`, but now I think that actually doesn't work, since it would still need to cast the argument to the unnameable interface type internally. _Sigh_, Go and/or protobuf.

Anyway, that was a long way of explaining the conclusion that I think the caller _does_ need to construct the `experimentv1.SearcherEvent` itself, incredibly annoyingly. But you can just not put anything for the `Id
` field (which will set it to 0).

Also `Enqueue` doesn't need to return an error anymore.